util/blockdevice/topology.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2020 The MayaData Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package blockdevice
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"mayadata.io/cstorpoolauto/types"
+)
+
+const (
+	// DeviceKindSSD represents device kind of SSD devices.
+	DeviceKindSSD string = "SSD"
+	// DeviceKindHDD represents device kind of HDD devices.
+	DeviceKindHDD string = "HDD"
+	// DeviceKindUnKnown represents unknown device type.
+	DeviceKindUnKnown string = "Unknown"
+)
+
+// MetaInfo contains identity of block device and some
+// meta information for a block device.
+type MetaInfo struct {
+	NodeName string
+	HostName string
+	Identity *types.Reference
+	Capacity *resource.Quantity
+}
+
+/*
+GetTopologyMapGroupByDeviceTypeAndBlockSize function builds a map using
+list of block devices. Block devices are grouped by device kind and blocksize
+ie - HDD, SSD-16384, HDD-4096, HDD-512 or SSD, SSD-4096, SSD-16384 and in
+one group all the block devices are arranged by node name.
+
+Top level key is contains DeviceType-PhysicalSectorSize
+
+1. initial stage it will be a list of block devices.
+[]blockdevice
+
+2. At last it will be converted into below map -
+
+- HDD:
+	- node-1 []blockdevice
+	- node-3 []blockdevice
+- HDD-4096:
+	- node-1 []blockdevice
+	- node-2 []blockdevice
+- HDD-16384:
+	- node-2 []blockdevice
+	- node-3 []blockdevice
+- SSD:
+	- node-1 []blockdevice
+	- node-2 []blockdevice
+- SSD-4096:
+	- node-1 []blockdevice
+	- node-2 []blockdevice
+- SSD-16384:
+	- node-1 []blockdevice
+	- node-3 []blockdevice
+
+*/
+func GetTopologyMapGroupByDeviceTypeAndBlockSize(
```

Comment:
It will be good if client has following code:

t := NewTopology(..)
t.GroupByDeviceTypeAndBlockSize()

Reply:
This part I will fix in some other PR